### DISCUSSION TITLE
Summary of tainted resources is wrong

### DISCUSSION THREAD

**Discussion by kyontan** [suzuki-shunsuke/tfcmt#823] [Deprecated (Bug Report)] [answered]
### tfcmt version

v4.4.1 (latest as of 2023-06-20)

### Environment

* OS: Ubuntu 22.04
* CPU Architecture: amd64

### Overview

When I taint resources manually, then the plan summary is shown a bit wrongly.

```
# terraform taint aws_rds_cluster.postgres_temporal
Resource instance aws_rds_cluster.postgres_temporal has been marked as tainted.
```

reference: https://developer.hashicorp.com/terraform/cli/state/taint

The bare terraform output is:
```
  # aws_rds_cluster.postgres_temporal is tainted, so must be replaced
-/+ resource "aws_rds_cluster" "postgres_temporal" {
...
}
```

Then the tfcmt result is shown like:

```
- Replace
    - aws_rds_cluster.postgres_temporal is tainted, so
```

<img width="696" alt="image" src="https://github.com/suzuki-shunsuke/tfcmt/assets/802339/5068c600-be77-4206-a2d1-6e4b1883122d">

### How to reproduce

I think this is just an omission of consideration of the case, so the `tfcmt.yaml` or other things are not related.

We may reproduce this by using null_resource.
(I don't have time for now, so I will append it later if needed)

### Debug output

```console
$ 
```

### Expected behaviour

tfcmt posts the summary like:

```
- Replace
    - aws_rds_cluster.postgres_temporal
```

### Actual behaviour

tfcmt posts the summary like:

```
- Replace
    - aws_rds_cluster.postgres_temporal is tainted, so
```

### Important Factoids

I think fixing the regular expression is enough

- https://github.com/suzuki-shunsuke/tfcmt/blob/f9b14230fd97a428194175a0db6b307cb60d2de1/pkg/terraform/parser.go#LL70C1-L70C1

**Comment by suzuki-shunsuke** [comment]
Thank you for your report!

**Reply by suzuki-shunsuke** [reply]
https://developer.hashicorp.com/terraform/cli/commands/taint

https://github.com/google/re2/wiki/Syntax

**Reply by suzuki-shunsuke** [reply]
- https://github.com/suzuki-shunsuke/tfcmt/pull/824

**Reply by suzuki-shunsuke** [reply]
I could reproduce the issue.

```console
$ terraform version
Terraform v1.5.0
on darwin_arm64
+ provider registry.terraform.io/hashicorp/null v3.2.1
```

main.tf

```tf
resource "null_resource" "foo" {}
```

Create a resource.

```console
$ terraform init
$ terraform apply
```

Run taint.

```console
$ terraform taint null_resource.foo
Resource instance null_resource.foo has been marked as tainted.
```

Run tfcmt plan.

```console
$ tfcmt plan -- terraform plan
```

---

## Plan Result

### :warning: Resource Deletion will happen :warning:
This plan contains resource delete operation. Please check the plan result very carefully!

<pre><code>Plan: 1 to add, 0 to change, 1 to destroy.</code></pre>

* Replace
  * null_resource.foo is tainted, so


<details><summary>Change Result (Click me)</summary>

```hcl

  # null_resource.foo is tainted, so must be replaced
-/+ resource "null_resource" "foo" {
      ~ id = "2678281950199183155" -> (known after apply)
    }

Plan: 1 to add, 0 to change, 1 to destroy.
```

</details>

**Reply by suzuki-shunsuke** [reply]
@kyontan 
Would you try v4.4.2-1?
https://github.com/suzuki-shunsuke/tfcmt/releases/tag/v4.4.2-1
I think this version solves the issue.

- https://github.com/suzuki-shunsuke/tfcmt/pull/824

If there is no problem, I'll release v4.4.2.

**Reply by suzuki-shunsuke** [reply, marked as answer]
tfcmt v4.4.2 is out 🎉 
The issue has been solved.
https://github.com/suzuki-shunsuke/tfcmt/releases/tag/v4.4.2
